### PR TITLE
Warn users with harmful `path_helper` calls

### PR DESCRIPTION
**Pull request by derekprior**
El Capitan moves the `path_helper` call to `zprofile`, but our dotfiles
had already accounted for it being in `zshenv` where it was previously.

### CODE REVIEW

**Comment by geoffharcourt** [comment]
Correctly warned on my machine.

**Comment by geoffharcourt** [comment]
The warning message is missing the word `path_helper` on my screen output.

**Comment by derekprior** [comment]
That's because I was so helpfully calling path_helper rather than displaying it.

**Comment by geoffharcourt** [comment]
One more issue: if `zprofile` doesn't exist, there's an error message:

```shell
❯ rcup
grep: /etc/zprofile: No such file or directory
```

**Comment by derekprior** [comment]
hm. I thought that's what `--silet` was supposed to fix.

File: hooks/post-up [hunk 1]
```diff
@@ -6,4 +6,15 @@ if [ ! -e $HOME/.vim/autoload/plug.vim ]; then
   curl -fLo $HOME/.vim/autoload/plug.vim --create-dirs \
       https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
 fi
+
 vim -u $HOME/.vimrc.bundles +PlugInstall +PlugClean! +qa
+
+if grep --quiet --silent 'path_helper' /etc/zprofile; then
```
Comment (gabebw) [inline hooks/post-up]: According to `man grep` on OS X, `--quiet` and `--silent` do the same thing:

```
 -q, --quiet, --silent
             Quiet mode: suppress normal output.  grep will only search a file until a match has been found, making searches potentially less expensive.
```

Do you mean `--quiet --no-messages`?

```
     -s, --no-messages
             Silent mode.  Nonexistent and unreadable files are ignored (i.e. their error messages are suppressed).
```

Is there a difference between OS X, *BSD, and Linux grep here? @thoughtbot/shell

**Comment by derekprior** [comment]
Updated. The flag I wanted was `--no-messages`

File: hooks/post-up [hunk 2]
```diff
@@ -6,4 +6,15 @@ if [ ! -e $HOME/.vim/autoload/plug.vim ]; then
   curl -fLo $HOME/.vim/autoload/plug.vim --create-dirs \
       https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
 fi
+
 vim -u $HOME/.vimrc.bundles +PlugInstall +PlugClean! +qa
+
+if grep --quiet --no-messages 'path_helper' /etc/zprofile; then
```
Comment (gabebw) [inline hooks/post-up]: Technically this doesn't require the quotes. But it doesn't really matter.

**Comment by gabebw** [comment]
I have no further input on the quality of this shell script.

File: hooks/post-up [hunk 3]
```diff
@@ -6,4 +6,15 @@ if [ ! -e $HOME/.vim/autoload/plug.vim ]; then
   curl -fLo $HOME/.vim/autoload/plug.vim --create-dirs \
       https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
 fi
+
 vim -u $HOME/.vimrc.bundles +PlugInstall +PlugClean! +qa
+
+if grep --quiet --no-messages path_helper /etc/zprofile; then
```
Comment (mike-burns) [inline hooks/post-up]: The more portable way to write this is:

```sh
if grep path_helper /etc/zprofile >/dev/null 2>/dev/null; then
```

That will work on unix systems in addition to GNU.

**Comment by derekprior** [comment]
Looks like we're going in another direction.

**Comment by geoffharcourt** [comment]
Hi @derekprior I wanted to wait until we resolved the issues in #426 before making a call here.

**Comment by derekprior** [comment]
In any event - the code there will be more useful in displaying a better message.

**Pull request closed without merging**